build-system/tasks/cherry-pick.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const argv = require('minimist')(process.argv.slice(2));
+const log = require('fancy-log');
+const {getOutput} = require('../common/exec');
+const {green, cyan, red, yellow} = require('ansi-colors');
+
+/**
+ * Executes a shell command, and logs an error message if the command fails.
+ *
+ * @param {string} cmd
+ * @param {string} msg
+ * @return {!Object}
+ */
+function execOrLog(cmd, msg) {
+  const result = getOutput(cmd);
+  if (result.status) {
+    log(yellow('ERROR:'), msg);
+    throw new Error(result.stderr);
+  }
+
+  return result;
+}
+
+/**
+ * Determines the name of the cherry-pick branch.
+ *
+ * @param {string} version
+ * @return {string}
+ */
+function cherryPickBranchName(version) {
+  const timestamp = version.slice(0, -3);
+  let suffix = String(Number(version.slice(-3)) + 1);
+  while (suffix.length < 3) {
+    suffix = `0${suffix}`;
+  }
+  return `amp-release-${timestamp}${suffix}`;
+}
+
+/**
+ * Updates tags from the remote and creates a branch at the release commit.
+ *
+ * @param {string} ref
+ * @param {!Array<string>} commits
+ * @param {string} branch
+ * @param {string} remote
+ */
+function prepareBranch(ref, commits, branch, remote) {
+  const needsFetch = [ref]
+    .concat(commits)
+    .some((r) => getOutput(`git rev-parse ${r}`).status);
+
+  if (needsFetch) {
+    log(green('INFO:'), 'Fetching latest tags and commits from', cyan(remote));
+    execOrLog(
+      `git fetch ${remote}`,
+      `Failed to fetch updates from remote ${cyan(remote)}`
+    );
+  } else {
+    log(
+      green('INFO:'),
+      'Identified tag and all commits available in local repository'
+    );
+  }
+
+  execOrLog(
+    `git checkout -b ${branch} ${ref}`,
+    `Failed to checkout new branch at ref ${cyan(ref)}`
+  );
+}
+
+/**
+ * Cherry-picks a commit into a new branch. When the cherry-pick succeeds,
+ * returns `true`. In the event of a merge conflict, the cherry-pick is aborted
+ * and an error is thrown.
+ *
+ * @param {string} sha
+ */
+function performCherryPick(sha) {
+  try {
+    log(green('INFO:'), 'Cherry-picking commit', cyan(sha));
+    execOrLog(
+      `git cherry-pick -x ${sha}`,
```

Comment:
`git cherry-pick` can take multiple SHAs and would fail the whole command if any one of them causes a merge conflict, instead of leaving the workspace in a partial state. This should be changed to `git cherry-pick ${shas.join(' ')}` :)

Reply:
See https://github.com/ampproject/amphtml/pull/28614#discussion_r432578347